File: backend/Models/Place.cs
using System.Text.Json.Serialization;

namespace ExploreHKMOApi.Models;

public class Place
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Region { get; set; } = null!;
    public string Category { get; set; } = null!;
    public Description Description { get; set; } = default!;

    public Location Location { get; set; } = null!;
    public Hours? Hours { get; set; }
    public List<string> Images { get; set; } = new List<string>();

    public double? Rating { get; set; }
    public int? Ranking { get; set; }
    public string? Phone { get; set; }
    public string? Website { get; set; }

    [JsonPropertyName("_connection")]
    public Connection? Connection { get; set; }
}

public class Description
{
    public string Content { get; set; } = "";
    public string Source { get; set; } = "tripadvisor"; //tripadvisor|google-map|ai
}

public class Hours
{
    public string Timezone { get; set; } = "Asia/Hong_Kong";
    public List<RegularHour> Regular { get; set; } = new();
    public List<HourException>? Exceptions { get; set; }
}

public class RegularHour
{
    public int Day { get; set; }       // 1–7 Monday to Sunday
    public string Open { get; set; } = "";
    public string Close { get; set; } = "";
}

public class HourException
{
    public string Date { get; set; } = ""; // "YYYY-MM-DD"
    public bool? Closed { get; set; }
    public string? Open { get; set; }
    public string? Close { get; set; }
    public string? Notes { get; set; }
}

public class Location
{
    public string Address { get; set; } = "";
    public double Latitude { get; set; }
    public double Longitude { get; set; }
}

public class Connection
{
    public string Type { get; set; } = "tripadvisor";
    public int LocationId { get; set; }
}

File: scripts/models/place.py
import json
from dataclasses import dataclass, field, asdict
import re
from typing import List, Optional


@dataclass
class Description:
    content: str = ""
    source: str = "tripadvisor"  # tripadvisor|google-map|ai


@dataclass
class RegularHour:
    day: int  # 1–7 Monday to Sunday
    open: str = ""
    close: str = ""


@dataclass
class HourException:
    date: str = ""  # "YYYY-MM-DD"
    closed: Optional[bool] = None
    open: Optional[str] = None
    close: Optional[str] = None


@dataclass
class Hours:
    timezone: str = "Asia/Hong_Kong"
    regular: List[RegularHour] = field(default_factory=list)
    exceptions: Optional[List[HourException]] = None


@dataclass
class Location:
    address: str = ""
    latitude: Optional[float] = None
    longitude: Optional[float] = None


@dataclass
class Connection:
    type: str  # tripadvisor
    locationId: int


@dataclass
class Place:
    id: int
    name: str
    region: str
    category: str
    description: Description
    location: Location
    images: List[str]
    hours: Optional[Hours] = None
    rating: Optional[float] = None
    ranking: Optional[int] = None
    phone: Optional[str] = None
    website: Optional[str] = None
    _connection: Optional[Connection] = None

    def to_dict(self):
        return asdict(self)

    def to_json(self) -> str:
        return re.sub(
            r'\{\s*"day":\s*(\d+),\s*"open":\s*"([^"]*)",\s*"close":\s*"([^"]*)"\s*\}',
            r'{ "day": \1, "open": "\2", "close": "\3" }',
            json.dumps(self.to_dict(), indent=2, ensure_ascii=False),
            flags=re.MULTILINE | re.DOTALL,
        )
